### PR TITLE
fix(useAccordion2): Ignore changes to defaultOpen

### PR DESCRIPTION
**Pull request by lukelooker**
Currently `useAccordion2` (and by proxy `Accordion2` & `Accordion`) responds to mutations of the `defaultOpen` property. Instead it should be a one-and-done property – changes should be ignored.

### CODE REVIEW

File: packages/components/src/Accordion2/Accordion2.spec.tsx [hunk 1]
```diff
@@ -111,4 +111,25 @@ describe('Accordion2', () => {
...
+          </Accordion2>
+          )
+        </>
+      )
+    }
+
+    renderWithTheme(<Example />)
+    expect(screen.getByText('World')).toBeInTheDocument()
+    fireEvent.click(screen.getByText('Toggle'))
+    expect(screen.queryByText('World')).not.toBeInTheDocument()
```
Comment (ElliotTheHuman) [inline packages/components/src/Accordion2/Accordion2.spec.tsx:133]: Dumb question, but we want `defaultOpen` to NOT change the open state after initial render right?

In which case, we'd want this line of the test to also be, I think?
```
expect(screen.getByText('World')).toBeInTheDocument()
```

File: packages/components/src/Accordion2/Accordion2.spec.tsx [hunk 2]
```diff
@@ -111,4 +111,25 @@ describe('Accordion2', () => {
...
+    const Example = () => {
+      const [defaultOpen, setDefaultOpen] = React.useState(true)
+
+      return (
+        <>
+          <button onClick={() => setDefaultOpen(!defaultOpen)}>Toggle</button>
+          <Accordion2 defaultOpen={defaultOpen} label="Hello">
+            World
+          </Accordion2>
+          )
```
Comment (ElliotTheHuman) [inline packages/components/src/Accordion2/Accordion2.spec.tsx:125]: I don't think this closing parenthesis has a corresponding open one. Should be able to just delete it.

File: packages/components/src/Accordion2/useAccordion2.tsx [hunk 3]
```diff
@@ -59,7 +59,13 @@ export const useAccordion2 = ({
   toggleOpen: propsToggleOpen,
   ...props
 }: Accordion2Props) => {
-  const { value, toggle } = useToggle(!!defaultOpen)
+  const { change, value, toggle } = useToggle(false)
+  useEffect(() => {
+    if (defaultOpen !== undefined) {
+      change(defaultOpen)
+    }
+  })
```
Comment (ElliotTheHuman) [inline packages/components/src/Accordion2/useAccordion2.tsx:67]: Should be able to get the desired "one and done" behavior by passing an empty array in as the second argument of this `useEffect`.

**Review by ElliotTheHuman — CHANGES_REQUESTED** [review]

File: packages/components/src/Accordion2/useAccordion2.tsx (hunk 3 above)
Comment (lukelooker) [reply packages/components/src/Accordion2/useAccordion2.tsx]: Doh!

File: packages/components/src/Accordion2/Accordion2.spec.tsx (hunk 2 above)
Comment (lukelooker) [reply packages/components/src/Accordion2/Accordion2.spec.tsx:125]: Done! :)

**Review by ElliotTheHuman — APPROVED** [review]
LGTM!!

**Review by ElliotTheHuman — CHANGES_REQUESTED** [review]
Looking at the image snapshot / storybook errors, I think the issue is that the `Tree` / `Accordion` will always initially render as closed, but after that initial render the `useEffect` will fire and then change the state to open. So in snapshot land, you get this weird effect where the accordions / trees start as closed but quickly switch to open (eye tests in storybook support this).

Ideally, we'd just want that initial render to know that it should be open, so maybe a viable option would be to have the existing `defaultOpen` implementation, and then add a state piece that checked for the initial render. If the initial render happened, then use the added state to ignore the `defaultOpen` on subsequent rerenders.

**Review by ElliotTheHuman — APPROVED** [review]
LGTM

**Pull request merged**